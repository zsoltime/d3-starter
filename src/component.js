const addBackground = (selection, { height, width }) => {
  const background = selection.selectAll('rect').data([null]);

  background
    .enter()
    .append('rect')
    .attr('fill', '#edf6f0')
    .merge(background)
    .attr('height', height)
    .attr('width', width);
};

const addText = (selection, { height, width }) => {
  const text = selection.selectAll('text').data([null]);

  text
    .enter()
    .append('text')
    .text('D3.js Starter')
    .attr('alignment-baseline', 'middle')
    .attr('text-anchor', 'middle')
    .style('font-size', '10vmin')
    .style('font-weight', '100')
    .merge(text)
    .attr('transform', `translate(${width / 2}, ${height / 2})`);
};

const component = (selection, props) => {
  const svg = selection.selectAll('svg').data([null]);

  const svgMerged = svg
    .enter()
    .append('svg')
    .merge(svg)
    .attr('height', props.height)
    .attr('width', props.width);

  addBackground(svgMerged, props);
  addText(svgMerged, props);
};

export default component;
