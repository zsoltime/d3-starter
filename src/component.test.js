import { select } from 'd3';

import component, { addBackground, addText } from './component';

describe('component', () => {
  const body = select('body');
  let svg = null;

  beforeEach(() => {
    svg = body.append('svg');
  });

  afterEach(() => {
    select('svg').remove();
  });

  describe('addBackground', () => {
    it('matches snapshot', () => {
      addBackground(svg, { height: 400, width: 600 });
      expect(svg._groups[0][0]).toMatchSnapshot();
    });
  });

  describe('addText', () => {
    it('matches snapshot', () => {
      addText(svg, { height: 400, width: 600 });

      expect(svg._groups[0][0]).toMatchSnapshot();
    });
  });

  describe('component', () => {
    it('matches snapshot', () => {
      component(body, { height: 400, width: 600 });

      expect(body._groups[0][0]).toMatchSnapshot();
    });
  });
});
