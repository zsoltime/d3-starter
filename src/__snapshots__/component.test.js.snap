// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component addBackground matches snapshot 1`] = `
<svg>
  <rect
    fill="#edf6f0"
    height="400"
    width="600"
  />
</svg>
`;

exports[`component addText matches snapshot 1`] = `
<svg>
  <text
    alignment-baseline="middle"
    style="font-size: 10vmin; font-weight: 100;"
    text-anchor="middle"
    transform="translate(300, 200)"
  >
    D3.js Starter
  </text>
</svg>
`;

exports[`component component matches snapshot 1`] = `
<body>
  <svg
    height="400"
    width="600"
  >
    <rect
      fill="#edf6f0"
      height="400"
      width="600"
    />
    <text
      alignment-baseline="middle"
      style="font-size: 10vmin; font-weight: 100;"
      text-anchor="middle"
      transform="translate(300, 200)"
    >
      D3.js Starter
    </text>
  </svg>
</body>
`;
